client/landing/stepper/declarative-flow/free.ts

```diff
@@ -151,6 +139,54 @@ const free: Flow = {
 
 		return { goNext, goBack, goToStep, submit };
 	},
+
+	useAssertConditions(): AssertConditionResult {
+		const userIsLoggedIn = useSelect( ( select ) => select( USER_STORE ).isCurrentUserLoggedIn() );
```

Comment:
Moved free flow redirect logic here since it was removed above when I removed the intro page-related code. This will ensure users are logged in before they can interact with the free flow.

Reply:
I think I might not be comprehending why this logic needed to be moved? Couldn't we have shifted the `intro` page-related code into the `freeSetup` switch statement?

Reply:
Example:

```javascript
case 'freeSetup':
  clearSignupDestinationCookie();
  if ( userIsLoggedIn ) {
    return navigate( 'siteCreationStep' );
  }
  return window.location.assign( logInUrl );
```

Reply:
On production, you are able to access the other screens in the free flow (`freeSetup` & `designSetup`) while you're logged out. Since I was moving this logic as I removed the intro screen related code, I figured I would also fix the logged/logged in redirection for all the free flow screens.

Reply:
I was wondering the same thing and had to look up the useAssertConditions() hook, which has a good explanation in the readme. I think what this does is provide a more robust solution that run the checks for all steps in the flow instead of just one.

Reply:
I see! Thanks for clarifying. For others reading this, there are similar conventions already established in other flows like `ecommerce` https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/declarative-flow/tailored-ecommerce-flow.ts#L51-L97

Reply:
Should note that's for all the conditional checks and redirecting to the login url.  Moving the the clearSignupDestinationCookie() call seems a bit separate from that. I'd think that invoking that could work here, within the switch for the setup step, or in the setup step itself as done in this PR.

Reply:
The hook name was also throwing me off quite a bit, but, for others, there's documentation in stepper describing `useAssertConditions` https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/README.md

Reply:
If we do go with this approach, because we need to run similar checks for logged in users for the other flows, it may be worth finding a way to extract this logic to something re-usable so we don't have to duplicate all the logic here. But that's something I'd do in a separate PR.